[ADAM-510] Refactoring RDD function names

heuermh opened the pull request:
Fixes #910 

Summary of initial changes:
```
class ADAMContext:

adamBamDictionaryLoad  --> loadBamDictionary
adamBamDictionaryLoad --> loadBamDictionary
adamBamLoadReadGroups --> loadBamReadGroups
loadParquet
adamDictionaryLoad --> loadDictionary
loadBam
loadIndexedBam
loadAvro
loadParquetAlignments
loadInterleavedFastq
loadFastq
loadPairedFastq
loadUnpairedFastq
loadVcf
loadParquetGenotypes
loadParquetVariants
loadFasta
loadInterleavedFastqAsFragments
loadGTF --> loadGtf
loadBED --> loadBed
loadNarrowPeak
loadIntervalList
loadParquetFeatures
loadParquetContigFragments
loadParquetFragments
loadVcfAnnotations
loadParquetVariantAnnotations
loadVariantAnnotations
loadFeatures
loadGenes
loadReferenceFile
loadSequence --> loadSequences
loadGenotypes
loadVariants
loadAlignments
loadFragments
loadAlignmentsFromPaths
findFiles

class ADAMRDDFunctions:

adamParquetSave --> saveAsParquet
adamParquetSave --> saveAsParquet

class ADAMSequenceDictionaryRDDAggregator:

getSequenceRecordsFromElement --> getSequenceRecords
adamGetSequenceDictionary --> getSequenceDictionary

class ADAMSpecificRecordSequenceDictionaryRDDAggregator:

getSequenceRecordsFromElement --> getSequenceRecords

object PairingRDD:

rddToPairingRDD

class NucleotideContigFragmentRDDFunctions:

toReads
saveAsFasta
mergeFragments
adamGetReferenceString --> getReferenceString
getSequenceRecordsFromElement --> getSequenceRecords
flankAdjacentFragments
countKmers

class FeatureRDDFunctions:

asGenes --> toGenes
filterByOverlappingRegion

class FragmentRDDFunctions:

toReads
getSequenceRecordsFromElement --> getSequenceRecords

class AlignmentRecordRDDFunctions:

filterByOverlappingRegion
saveAsParquet
adamAlignedRecordSave --> remove method
adamSave --> save
adamSAMString --> saveAsSamString
adamSAMSave --> saveAsSam
getSequenceRecordsFromElement --> getSequenceRecords
adamConvertToSAM --> convertToSam
adamCountKmers --> countKmers
adamSortReadsByReferencePosition --> sortReadsByReferencePosition
adamMarkDuplicates --> markDuplicates
adamBQSR --> bqsr
adamRealignIndels --> realignIndels
adamFlagStat --> flagStat
adamSingleReadBuckets --> groupReads
adamCharacterizeTags --> characterizeTags
adamCharacterizeTagValues --> characterizeTagValues
adamFilterRecordsWithTag --> filterRecordsWithTag
adamSaveAsPairedFastq --> saveAsPairedFastq
adamSaveAsFastq --> saveAsFastq
adamRePairReads --> reassembleReadPairs
toFragments

class VariantContextRDDFunctions:

getSequenceRecordsFromElement --> getSequenceRecords
joinDatabaseVariantAnnotation
getCallsetSamples
saveAsVcf

class GenotypeRDDFunctions:

toVariantContext
filterByOverlappingRegion
```

Will take a closer look at scaladoc strings and other parts of the codebase in future commits.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1120/<h2>Build result: FAILURE</span></h2>GitHub pull request #979 of commit febc58c133ffb61b707aa76ea8f38e6525f640d8 automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse origin/pr/979/merge^{commit} # timeout=10 > git branch -a --contains 1b4c17c6edaae3d5ee128816570e3bb454dd9c6a # timeout=10 > git rev-parse remotes/origin/pr/979/merge^{commit} # timeout=10Checking out Revision 1b4c17c6edaae3d5ee128816570e3bb454dd9c6a (origin/pr/979/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f 1b4c17c6edaae3d5ee128816570e3bb454dd9c6aFirst time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

fnothaft commented:
You are a true patriot, @heuermh.

+1

heuermh commented:
I've hit some difference between the maven scala compiler and the IDEA one though, I'm afraid.
```
[ERROR] ./adam-core/src/.../AlignmentRecordRDDFunctions.scala:199:
  error: recursive method saveAsParquet needs result type
[ERROR]     rdd.saveAsParquet(args)
```
Is there some way I can force a certain implicit?  This call should be to ```ADAMRDDFunctions.saveAsParquet(args: SaveArgs)```, not recursive on itself.

heuermh commented:
For lack of a better way around the problem above, I've made ```AlignmentRecordRDDFunctions.saveAsParquet``` private and renamed it to ```saveParquet```, similar to ```AlignmentRecordRDDFunctions.saveAvro```.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1121/
Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1123/
Test PASSed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1124/<h2>Build result: FAILURE</span></h2>GitHub pull request #979 of commit 30f979a634b2614e2109ceecbc5d81e256cf5abc automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse origin/pr/979/merge^{commit} # timeout=10 > git branch -a --contains 3eb17d512560267a9ad144d1fb03272899187a2d # timeout=10 > git rev-parse remotes/origin/pr/979/merge^{commit} # timeout=10Checking out Revision 3eb17d512560267a9ad144d1fb03272899187a2d (origin/pr/979/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f 3eb17d512560267a9ad144d1fb03272899187a2dFirst time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1126/<h2>Build result: FAILURE</span></h2>GitHub pull request #979 of commit 6f55f6a7d71e5c3e72c0aeecf8439926a9062b2d automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse origin/pr/979/merge^{commit} # timeout=10 > git branch -a --contains bd6707a304521cf43450c334cae2144e60260e49 # timeout=10 > git rev-parse remotes/origin/pr/979/merge^{commit} # timeout=10Checking out Revision bd6707a304521cf43450c334cae2144e60260e49 (origin/pr/979/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f bd6707a304521cf43450c334cae2144e60260e49First time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1134/
Test PASSed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1136/<h2>Build result: FAILURE</span></h2>GitHub pull request #979 of commit 3a7f598d4490c54e52ec0d0cbe2d435b0cb794d2 automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse origin/pr/979/merge^{commit} # timeout=10 > git branch -a --contains 690bd2ffa5769c8237c5c8c112647fb7dbf50998 # timeout=10 > git rev-parse remotes/origin/pr/979/merge^{commit} # timeout=10Checking out Revision 690bd2ffa5769c8237c5c8c112647fb7dbf50998 (origin/pr/979/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f 690bd2ffa5769c8237c5c8c112647fb7dbf50998First time build. Skipping changelog.Triggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

heuermh commented:
Jenkins, retest this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1137/
Test PASSed.

heuermh commented:
@fnothaft @jpdna @tdanford @ryan-williams this one is ready to review.  I know the new unit tests have weak assertions; I wasn't convinced it'd be worth round-tripping them all.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1150/<h2>Build result: FAILURE</span></h2>[...truncated 24 lines...]Triggering ADAM-prb ? 2.6.0,2.10,1.3.1,centosTriggering ADAM-prb ? 2.6.0,2.10,1.6.0,centosTriggering ADAM-prb ? 2.6.0,2.10,1.4.1,centosTriggering ADAM-prb ? 2.6.0,2.11,1.3.1,centosTriggering ADAM-prb ? 2.3.0,2.10,1.4.1,centosTriggering ADAM-prb ? 2.3.0,2.10,1.3.1,centosTriggering ADAM-prb ? 2.3.0,2.11,1.4.1,centosTriggering ADAM-prb ? 2.3.0,2.11,1.3.1,centosTriggering ADAM-prb ? 2.3.0,2.10,1.5.2,centosADAM-prb ? 2.6.0,2.11,1.6.0,centos completed with result SUCCESSADAM-prb ? 2.3.0,2.11,1.5.2,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,1.4.1,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.6.0,centos completed with result FAILUREADAM-prb ? 2.3.0,2.11,1.6.0,centos completed with result FAILUREADAM-prb ? 2.6.0,2.10,1.3.1,centos completed with result FAILUREADAM-prb ? 2.6.0,2.10,1.6.0,centos completed with result FAILUREADAM-prb ? 2.6.0,2.10,1.4.1,centos completed with result SUCCESSADAM-prb ? 2.6.0,2.11,1.3.1,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.4.1,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.3.1,centos completed with result FAILUREADAM-prb ? 2.3.0,2.11,1.4.1,centos completed with result FAILUREADAM-prb ? 2.3.0,2.11,1.3.1,centos completed with result FAILUREADAM-prb ? 2.3.0,2.10,1.5.2,centos completed with result FAILURENotifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

heuermh commented:
Jenkins, retest this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1151/
Test PASSed.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala:
```diff
@@ -17,20 +17,15 @@
-}
+import java.io.{ InputStream, OutputStream, StringWriter, Writer }
+import java.lang.reflect.InvocationTargetException
+
```
Ixnay on the spaces.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala:
```diff
@@ -44,16 +39,19 @@ import org.bdgenomics.adam.models._
 import org.bdgenomics.adam.util.MapTools
 import org.bdgenomics.formats.avro._
 import org.seqdoop.hadoop_bam.SAMRecordWritable
+
```
Drop space.

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala:
```diff
@@ -672,7 +649,7 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord]) extends Serializabl
    * @return Returns an RDD of recalibrated reads.
    */
-  def adamBQSR(
+  def bqsr(
```
If we're renaming this, I might prefer `recalibateBaseQualities` over `bqsr`. Thoughts?

heuermh replied:
+1

fnothaft commented on adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDDFunctions.scala:
```diff
@@ -712,7 +689,7 @@ class AlignmentRecordRDDFunctions(rdd: RDD[AlignmentRecord]) extends Serializabl
    * @return SingleReadBuckets with primary, secondary and unmapped reads
    */
-  def adamSingleReadBuckets(): RDD[SingleReadBucket] = {
+  def groupReads(): RDD[SingleReadBucket] = {
```
`toSingleReadBuckets`?

heuermh replied:
I went with the doc comment, ```Groups all reads by record group and read name```.  I don't think the word Bucket belongs in software.

fnothaft replied:
> I went with the doc comment, `Groups all reads by record group and read name.`

Fair, but `groupReads` is a bit underspecified. I.e., is it grouping by:

* Read group
* Position
* Library
* Etc.

Since it is grouping reads together by the original sequencing fragment, perhaps `groupReadsByFragment`?

> I don't think the word Bucket belongs in software.

Hah! Why's that?
In any case, perhaps do you want to `s/SingleReadBucket/SingleRead/g` or something like that?

heuermh replied:
> Since it is grouping reads together by the original sequencing fragment, perhaps ```groupReadsByFragment```?

+1

> In any case, perhaps do you want to ```s/SingleReadBucket/SingleRead/g``` or something like that?

I think I'll wait, I may want to propose other schema/class name changes in a separate pull request.

fnothaft replied:
> I think I'll wait, I may want to propose other schema/class name changes in a separate pull request.

SGTM!

fnothaft commented:
Other than a few nits, LGTM!

heuermh commented:
Addressed review comments and squashed commits.

Could I get a few more +1s?  This is a non-binary compatible change that will probably break 99% of external code and scripts.

fnothaft commented:
> Could I get a few more +1s? This is a non-binary compatible change that will probably break 99% of external code and scripts.

Agreed. I am +1, but it would be good to get more eyes.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1154/<h2>Build result: FAILURE</span></h2>GitHub pull request #979 of commit a649f55e3839f3e677cf641a66f16bf070ff931a automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse origin/pr/979/merge^{commit} # timeout=10 > git branch -a --contains 031d36d024933dd66f817dfdae0bca8c431b1be2 # timeout=10 > git rev-parse remotes/origin/pr/979/merge^{commit} # timeout=10Checking out Revision 031d36d024933dd66f817dfdae0bca8c431b1be2 (origin/pr/979/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f 031d36d024933dd66f817dfdae0bca8c431b1be2 > git rev-list 86417b1695dc72ab989275a1c69d26f54411bb75 # timeout=10Triggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1155/<h2>Build result: FAILURE</span></h2>GitHub pull request #979 of commit c2174329c7cee36692813bb1c72ef8b6ad1bb525 automatically merged.Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'[EnvInject] - Loading node environment variables.Building remotely on amp-jenkins-worker-05 (centos spark-test) in workspace /home/jenkins/workspace/ADAM-prb > git rev-parse --is-inside-work-tree # timeout=10Fetching changes from the remote Git repository > git config remote.origin.url https://github.com/bigdatagenomics/adam.git # timeout=10Fetching upstream changes from https://github.com/bigdatagenomics/adam.git > git --version # timeout=10 > git fetch --tags --progress https://github.com/bigdatagenomics/adam.git +refs/pull/*:refs/remotes/origin/pr/* > git rev-parse origin/pr/979/merge^{commit} # timeout=10 > git branch -a --contains e201a220b8af9bc5a7001296534c62af111b8bd7 # timeout=10 > git rev-parse remotes/origin/pr/979/merge^{commit} # timeout=10Checking out Revision e201a220b8af9bc5a7001296534c62af111b8bd7 (origin/pr/979/merge) > git config core.sparsecheckout # timeout=10 > git checkout -f e201a220b8af9bc5a7001296534c62af111b8bd7 > git rev-list 031d36d024933dd66f817dfdae0bca8c431b1be2 # timeout=10Triggering ADAM-prb ? 2.6.0,2.10,1.5.2,centosTriggering ADAM-prb ? 2.6.0,2.11,1.5.2,centosTouchstone configurations resulted in FAILURE, so aborting...Notifying endpoint 'HTTP:https://webhooks.gitter.im/e/ac8bb6e9f53357bc8aa8'
Test FAILed.

heuermh commented:
Jenkins, retest this please.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/ADAM-prb/1156/
Test PASSed.

tdanford commented:
+1

ryan-williams commented:
+1 thanks @heuermh

fnothaft commented:
Thanks to @heuermh for the PR, and to all for the review! I've merged as https://github.com/bigdatagenomics/adam/commit/70eaa70835c11fb81445a22a7b1752d831c66bd3.
